typing/typecore.ml

```diff
@@ -545,6 +545,40 @@ let has_local_attr_pat ppat =
 let has_local_attr_exp pexp =
   has_local_attr pexp.pexp_loc pexp.pexp_attributes
 
+let mode_cross env (ty : type_expr) mode =
+  if is_principal ty then begin
+    match immediacy env ty with
+    | Type_immediacy.Always -> Value_mode.newvar ()
+    | Type_immediacy.Always_on_64bits when !Clflags.native_code && Sys.word_size = 64 ->
+      begin
+        (* if !Clflags.verbose then begin
+         *   Format.eprintf "@[mode crossed!!@.%a@.@]" Printtyp.raw_type_expr  ty
+         * end; *)
+        Value_mode.newvar ()  (* floating and relaxed *)
+      end
+    | _ -> begin
+      (* if !Clflags.verbose then begin
+       *   Format.eprintf "@[mode not crossed because not immediate@.%a@.@]"
+       *     Printtyp.raw_type_expr ty
+       * end; *)
```

Comment:
Should this commented out code stay?

(I don't mind either way: leaving commented-out debug printing code can be useful if you think you'll need it soon, but if you didn't plan to keep it now is the time to delete it)

Reply:
FWIW I think debug code should always be deleted. You can find it in an old commit if you need it again, and when it isn't deleted it tends to stay forever slowly bit-rotting. For example, see the many commented out bits of code Jacques has left in the type-checker.